<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">

        <title>Accounting for Epistasis in PRSs Through the Coalescent</title>

        <link rel="stylesheet" href="dist/reset.css">
        <link rel="stylesheet" href="dist/reveal.css">
        <link rel="stylesheet" href="dist/theme/mytheme.css" id="theme">
    </head>
    <body>
        <div class="reveal">
            <div class="slides">
                <section class="cover">
                    <h1>Accounting for Epistasis in PRSs Through the Coalescent</h1>
                    <h2><a href="https://www.patrickfournier.ca">P. Fournier</a> &amp;
                        <a href="http://fabricelarribe.uqam.ca/">F. Larribe</a>
                        (<a href="https://statqam.uqam.ca/">STATQAM</a> &mdash; <a href="https://uqam.ca/">UQAM</a>)</h2>
                    <h3>New Statistical Methods in Genetic Studies</h3>
                    <h3>June 2<sup>nd</sup>, 2022</h3>
                </section>
                <section>
                    <section class="title-slide">
                        <h1>PRS &amp; Epistasis</h1>
                    </section>
                    <section data-markdown>
                        <textarea data-template>
# PRS: Definition
- "Weighted sum of the effects of a number of SNPs on a given quantitative
  trait"<span class="reference">1</span>
- Weights "based on the effect sizes of the SNPs estimated from a
  GWAS"<span class="reference">1</span>
- Originaly developped assuming **additive** effects<span class="reference">2</span>
---
# In practice
- PRS $\Leftrightarrow$ **linear combination** of counts of risk-increasing alleles

A PRS computed on $m$ alleles has the following form:
$$
\textcolor{#e65100}{\overbrace{\hat s}^{\text{score}}} = \sum\_{k = 1}^m
\textcolor{#4d7e65}{\underbrace{x_k}\_{\mathrlap{k-\text{allele count}}}}
\textcolor{#9558b2}{\overbrace{\hat{\beta}_k}^{\mathclap{\text{weight of allele }k}}}
$$
                        </textarea>
                    </section>
                    <section>
                        <h1>Assumptions</h1>
                        <ul>
                            <li>Weight estimation is based on <strong>linear models</strong></li>
                        </ul>

                        <div class="fragment fade-up">
                            <p>Assumes (among other things):</p>
                            <ul>
                                <li class="fragment highlight-current-red">Additivity</li>
                                <li class="fragment highlight-current-red grow">Linearity</li>
                            </ul>
                        </div>
                        <aside class="notes">
                            <ul>
                                <li>Additivity can be relaxed relatively easily (next slide);</li>
                                <li>It is much harder to let go of the linearity assumption;</li>
                                <li>Example of linear but non-additive model in next slide.</li>
                            </ul>
                        </aside>
                    </section>
                    <section data-markdown>
                        <textarea data-template>
# Non-additivity?
- Dominant, recessive, heterodominant, etc.
- Important for at least some
  phenotypes<span class="reference">3</span><sup>,</sup><span class="reference">4</span><sup>,</sup><span class="reference">5</span>

Example:<span class="reference">4</span> additive + dominant model:
$$
\hat s = \sum\_{k = 1}^m x_k \hat{\beta}_k +
\textcolor{#4d7e65}{\underbrace{d_k}\_{\mathclap{1 \text{ iff heterozygous at } k}}}
\textcolor{#9558b2}{\overbrace{\hat{\gamma}_k}^{\mathclap{\text{dominance weight of allele } k}}}
$$
                        </textarea>
                    </section>
                    <section>
                        <h1>Epistasis?</h1>
                        <ul>
                            <li>Gene-Gene interaction</li>
                            <li>Major difficulty in the analysis of GWAS data<span class="reference">6</span></li>
                            <li>Epistasis-aware models are possible; however, naïve ones are intractable ($\mathcal O(2^m)$ terms)</li>
                        </ul>

                        <div class="fragment fade-left">
                            <p>That being said,</p>
                            <ul>
                                <li>Some phenotypes are simple</li>
                                <li>Some forms of epistasis might be reflected in additive effects<span class="reference">7</span></li>
                            </ul>
                        </div>
                    </section>
                    <section>
                        <h1>Epistasis-aware models</h1>
                        <ul>
                            <li>Interaction learning<span class="reference">8</span><sup>,</sup><span class="reference">9</span></li>
                            <li>Machine learning</li>
                        </ul>

                        <div class="fragment fade-down">
                            <p>For machine learning:</p>
                            <ul>
                                <li>Marker selection is "the major factor that impacts on a machine learning model’s predictive performance"<span class="reference">10</span></li>
                                <li>The mechanic through which markers affect phenotype might not be known</li>
                            </ul>
                        </div>

                        <aside class="notes">
                            Two kinds of interaction learning:
                            <ol>
                                <li>Finding interactions associated with a given variable (begs the question: what does "associated" mean?);</li>
                                <li>Penalized regression: can still lead to explosion of # of variables.</li>
                            </ol>

                            Marker selection assumes a model of how genotype influence phenotype; this begs the question.
                        </aside>
                    </section>
                </section>
                <section>
                    <section class="title-slide">
                        <h1>Model-Free Genotype Based Prediction</h1>
                    </section>
                    <section>
                        <div class="r-stack">
                            <div class="fragment semi-fade-out" data-fragment-index="3">
                                <h1>Overview</h1>
                                Goal: compute the likelihood of phenotype $\varphi^*$ given the underlying genotype $h_0^*$.
                                <div class="fragment fade-down" data-fragment-index="1">
                                    Exploit information from paired haplotype-phenotype sequence:
                                    $$
                                    H_0 \bigtriangleup \Phi =
                                    \left\lbrace (h_0^1, \varphi_1), \ldots, (h_0^n, \varphi_n) \right\rbrace.
                                    $$
                                </div>
                                <div class="fragment fade-up" data-fragment-index="2">
                                    A bit of notation:
                                    $$
                                    H_0^* = H_0 \cup \lbrace h_0^* \rbrace, \quad
                                    \Phi^* = \Phi \cup \lbrace \varphi^* \rbrace
                                    $$
                                </div>
                            </div>
                            <div class="fragment r-frame obox fade-down" data-fragment-index="3">
                                <h2>Assumption: Unrelatedness</h2>
                                $$
                                \begin{align*}
                                h_0^i \perp h_0^j &\text{ where } h_0^i, h_0^j \in H_0^*\\
                                \varphi_i \perp \varphi_j &\text{ where } \varphi_i, \varphi_j \in \Phi^*
                                \end{align*}
                                $$
                            </div>
                        </div>
                    </section>
                    <section>
                        <div class="r-stack">
                            <div class="fragment semi-fade-out" data-fragment-index="2">
                                <h1>Likelihood</h1>
                                Assuming unrelatedness,
                                $$
                                L(\varphi^* | h_0^*, H_0, \Phi) \propto f(H_0^*, \Phi^*)
                                $$
                                <div class="fragment fade-up" data-fragment-index="1">
                                    The law of total probabilities allows the introduction of <strong>evolutionary history</strong> in the form of <span style="color: #4d7e65">genealogies</span>:
                                    $$
                                    f(H_0^*, \Phi^*)
                                    = \int_{\color{#4d7e65} \mathcal G} f(H_0^*, \Phi^* | {\color{#4d7e65} G}) g({\color{#4d7e65} G}) \text d{\color{#4d7e65} G}
                                    $$
                                </div>
                            </div>
                            <div class="fragment r-frame obox fade-down" data-fragment-index="2">
                                <h2>Assumption: Conditional Independance</h2>
                                $$
                                (H_0^* | {\color{#4d7e65} G}) \perp (\Phi^* | {\color{#4d7e65} G})
                                $$
                            </div>
                        </div>
                    </section>
                    <section data-markdown>
                        <textarea data-template>
# Likelihood &mdash; Conditional Independance
Assuming conditional independance,
$$
f(H_0^*, \Phi^*)
= \int_{\mathcal G} f(H_0^* | G) f(\Phi^* | G) g(G) \text d G
$$
---
# Consistency
- A genealogy $G$ is said to be consistent with a set of sequences $H_0$ if the sample induced by $G$ is precisely $H_0$
- A given genealogy is consistent with **exactly 1** sample

Thus,
$$
f(H_0^* | G) = \mathbb I(G \text{ consistent with } H_0^*),
$$
---
# Likelihood &mdash; Sample Consistency
Let $\mathcal G_{H_0^*} \subseteq \mathcal G$ be the set of genealogies consistent with $H_0^*$. The likelihood simplifies to
$$
L(\varphi^*| H_0^*, \Phi) \propto f(H_0^*, \Phi^*)
= \int_{\mathrlap{\mathcal G_{H_0^*}}} f(\Phi^* | G) g(G) \text d G
$$

Notes:
Conclusion:
- We avoid having to make assumptions about the mode of action of genotype on phenotype.
- On the other hand, we have to specify
  1. a distribution for genealogies;
  2. a conditional distribution for phenotypes.
                        </textarea>
                    </section>
                </section>
                <section>
                    <section class="title-slide">
                        <h1>Computing the Likelihood</h1>
                    </section>
                    <section>
                        <h1>Context</h1>
                        <ul>
                            <li>Discrete phenotype: $\Phi^* \in \lbrace 0, 1 \rbrace^{n + 1}$</li>
                            <li>Quantity of interest: $L(\varphi^* = 1 | H_0^*, \Phi)$</li>
                            <li>$G \sim$ ARG<span class="reference">11</span></li>
                            <li>$(\Phi^* | G) \sim$ ???</li>
                        </ul>
                        <span class="fragment fade-up"><p>First question: How to compute <strong>exact</strong> likelihood (not up to a constant)?</p></span>

                        <aside class="notes">
                            <ul>
                                <li>Flexible framework: main restriction is conditional independance between sequences and phenotypes.</li>
                                <li>Exact likelihood needed for PRS-like score.</li>
                            </ul>
                        </aside>
                    </section>
                    <section data-markdown>
                        <textarea data-template>
# Exact Likelihood
Efficient computation of the normalisation constant (Tonelli):
$$
\begin{align*}
f(H_0^*, \Phi)
&= {\color{#e65100} \int_{\mathrlap{\Omega_{\varphi^*}}}} f(\varphi^*, H_0^*, \Phi) \\;{\color{#e65100}\text d \varphi^*}\\\\
&= {\color{#e65100} \int_{\Omega_{\varphi^*}}} {\color{#9558b2} \int_{\mathrlap{\mathcal G_{H_0^*}}}} f(\varphi^*, \Phi | G) g(G) \\;{\color{#9558b2} \text d G} \\;{\color{#e65100} \text d \varphi^*}\\\\
&= {\color{#9558b2} \int_{\mathcal G_{H_0^*}}} {\color{#e65100} \int_{\mathrlap{\Omega_{\varphi^*}}}} f(\varphi^*, \Phi | G) g(G) \\;{\color{#e65100} \text d \varphi^*} \\;{\color{#9558b2} \text d G}
\end{align*}
$$

Notes:
The <span style="color: #e65100;">integral w.r.t φ<sup>*</sup></span> is often easier to compute than <span style="color: #9558b2;">the one w.r.t. G</span>. Example:
- Binary trait => sum of 2 terms;
- ARG => numerical integration.
                        </textarea>
                    </section>
                </section>
                <section id="references">
                    <section class="title-slide">
                        <h1>References</h1>
                    </section>
                    <section>
                        <ol>
                            <li>Croucha, D. J. M., &#38; Bodmer, W. F. (2020). Polygenic inheritance, GWAS, polygenic risk scores, and the search for functional variants. <i>Proceedings of the National Academy of Sciences of the United States of America</i>, <i>117</i>(32), 18924–18933. <a href="https://doi.org/10.1073/pnas.2005634117">https://doi.org/10.1073/pnas.2005634117</a></li>
                            <li>Meuwissen, T. H. E., Hayes, B. J., &#38; Goddard, M. E. (2001). Prediction of Total Genetic Value Using Genome-Wide Dense Marker Maps. <i>Genetics</i>, <i>157</i>(4), 1819–1829. <a href="https://doi.org/10.1093/GENETICS/157.4.1819">https://doi.org/10.1093/GENETICS/157.4.1819</a></li>
                            <li>Guindo-Martínez, M., et al. (2021). The impact of non-additive genetic associations on age-related complex diseases. <i>Nature Communications 2021 12:1</i>, <i>12</i>(1), 1–14. <a href="https://doi.org/10.1038/s41467-021-21952-4">https://doi.org/10.1038/s41467-021-21952-4</a></li>
                            <li>Pozarickij, A., Williams, C., &#38; Guggenheim, J. A. (2020). Non-additive (dominance) effects of genetic variants associated with refractive error and myopia. <i>Molecular Genetics and Genomics</i>, <i>295</i>(4), 843. <a href="https://doi.org/10.1007/S00438-020-01666-W">https://doi.org/10.1007/S00438-020-01666-W</a></li>
                            <li><i>Non Additive Genetic Effects Portal - Home</i>. (n.d.). Retrieved May 25, 2022, from <a href="https://nage.hugeamp.org/">https://nage.hugeamp.org/</a></li>
                        </ol>
                    </section>
                    <section>
                        <ol start="6">
                            <li>Furlong, L. I. (2013). Human diseases through the lens of network biology. <i>Trends in Genetics</i>, <i>29</i>(3), 150–159. <a href="https://doi.org/10.1016/J.TIG.2012.11.004">https://doi.org/10.1016/J.TIG.2012.11.004</a></li>
                            <li>Mäki-Tanila, A., &#38; Hill, W. G. (2014). Influence of Gene Interaction on Complex Trait Variation with Multilocus Models. <i>Genetics</i>, <i>198</i>(1), 355–367. <a href="https://doi.org/10.1534/GENETICS.114.165282">https://doi.org/10.1534/GENETICS.114.165282</a></li>
                            <li>Massi M.C., Franco N.R., Ieva F., Manzoni A., Paganoni A.M., Zunino P. HighOrder Interaction Learning via Targeted Pattern Search. <i>MOX Report 59/2020</i>, 2020. Retrieved May 25, 2022, from <a href="https://www.mate.polimi.it/biblioteca/add/qmox/59-2020.pdf">https://www.mate.polimi.it/biblioteca/add/qmox/59-2020.pdf</a></li>
                            <li>Franco, N. R., Massi, M. C., et al. (2021). Development of a method for generating SNP interaction-aware polygenic risk scores for radiotherapy toxicity. <i>Radiotherapy and Oncology</i>, <i>159</i>, 241–248. <a href="https://doi.org/10.1016/j.radonc.2021.03.024">https://doi.org/10.1016/j.radonc.2021.03.024</a></li>
                            <li>Ho, D. S. W., Schierding, W., Wake, M., Saffery, R., &#38; O’Sullivan, J. (2019). Machine learning SNP based prediction for precision medicine. <i>Frontiers in Genetics</i>, <i>10</i>(MAR), 267. <a href="https://doi.org/https://doi.org/10.3389/fgene.2019.00267">https://doi.org/https://doi.org/10.3389/fgene.2019.00267</a></li>
                        </ol>
                    </section>
                    <section>
                        <ol start="11">
                            <li>Griffiths, R. C., and Marjoram, P. (1996). An ancestral recombination graph. <i>IMA Volume on Mathematical Population Genetics</i> (P. Donnelly and S. Tavare, Eds.), Springer-Verlag, New York, 257–270</li>
                        </ol>
                    </section>
                </section>
            </div>
        </div>

        <div id="logo"><p>
            <a href="https://www.patrickfournier.ca">P. Fournier</a> &amp;
            <a href="http://fabricelarribe.uqam.ca/">F. Larribe</a>
            (<a href="https://statqam.uqam.ca/">STATQAM</a> &mdash; <a href="https://uqam.ca/">UQAM</a>)
            <span class="separator"></span>
            New Statistical Methods in Genetic Studies
            <span class="separator"></span>
            SSC Annual Meeting (Online)
            <span class="separator"></span>
            June 2<sup>nd</sup>, 2022
        </p></div>

        <div id="qrcode" alt="Slides"></div>

        <script src="dist/reveal.js"></script>
        <script src="plugin/notes/notes.js"></script>
        <script src="plugin/highlight/highlight.js"></script>
        <script src="plugin/math/math.js"></script>
        <script src="plugin/markdown/markdown.js"></script>
        <script src="node_modules/reveal.js-menu/menu.js"></script>
        <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.6.0/jquery.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/qrcodejs/1.0.0/qrcode.min.js" integrity="sha512-CNgIRecGo7nphbeZ04Sc13ka07paqdeTu0WR1IM4kNcpmBAUSHSQX0FslNhTDadL4O5SAGapGt4FodqL8My0mA==" crossorigin="anonymous" referrerpolicy="no-referrer"></script>
        <script src="dist/perso.js"></script>
        <script type="text/javascript">qrcode.makeCode("www.patrickfournier.ca/ssc2022");</script>
        <script>
         // More info about initialization & config:
         // - https://revealjs.com/initialization/
         // - https://revealjs.com/config/
         Reveal.initialize({
             hash: true,

             // Learn about plugins: https://revealjs.com/plugins/
             plugins: [ RevealHighlight, RevealMath.KaTeX, RevealMenu, RevealMarkdown, RevealNotes ],

             menu: {
                 // Specifies which side of the presentation the menu will
                 // be shown. Use 'left' or 'right'.
                 side: 'left',

                 // Specifies the width of the menu.
                 // Can be one of the following:
                 // 'normal', 'wide', 'third', 'half', 'full', or
                 // any valid css length value
                 width: 'normal',

                 // Add slide numbers to the titles in the slide list.
                 // Use 'true' or format string (same as reveal.js slide numbers)
                 numbers: false,

                 // Specifies which slide elements will be used for generating
                 // the slide titles in the menu. The default selects the first
                 // heading element found in the slide, but you can specify any
                 // valid css selector and the text from the first matching
                 // element will be used.
                 // Note: that a section data-menu-title attribute or an element
                 // with a menu-title class will take precedence over this option
                 titleSelector: 'h1, h2',

                 // If slides do not have a matching title, attempt to use the
                 // start of the text content as the title instead
                 useTextContentForMissingTitles: false,

                 // Hide slides from the menu that do not have a title.
                 // Set to 'true' to only list slides with titles.
                 hideMissingTitles: true,

                 // Adds markers to the slide titles to indicate the
                 // progress through the presentation. Set to 'false'
                 // to hide the markers.
                 markers: true,

                 // Specify custom panels to be included in the menu, by
                 // providing an array of objects with 'title', 'icon'
                 // properties, and either a 'src' or 'content' property.
                 custom: false,

                 // Specifies the themes that will be available in the themes
                 // menu panel. Set to 'true' to show the themes menu panel
                 // with the default themes list. Alternatively, provide an
                 // array to specify the themes to make available in the
                 // themes menu panel, for example...
                 //
                 // [
                 //     { name: 'Black', theme: 'dist/theme/black.css' },
                 //     { name: 'White', theme: 'dist/theme/white.css' },
                 //     { name: 'League', theme: 'dist/theme/league.css' },
                 //     {
                 //       name: 'Dark',
                 //       theme: 'lib/reveal.js/dist/theme/black.css',
                 //       highlightTheme: 'lib/reveal.js/plugin/highlight/monokai.css'
                 //     },
                 //     {
                 //       name: 'Code: Zenburn',
                 //       highlightTheme: 'lib/reveal.js/plugin/highlight/zenburn.css'
                 //     }
                 // ]
                 //
                 // Note: specifying highlightTheme without a theme will
                 // change the code highlight theme while leaving the
                 // presentation theme unchanged.
                 themes: false,

                 // Specifies the path to the default theme files. If your
                 // presentation uses a different path to the standard reveal
                 // layout then you need to provide this option, but only
                 // when 'themes' is set to 'true'. If you provide your own
                 // list of themes or 'themes' is set to 'false' the
                 // 'themesPath' option is ignored.
                 themesPath: 'dist/theme/',

                 // Specifies if the transitions menu panel will be shown.
                 // Set to 'true' to show the transitions menu panel with
                 // the default transitions list. Alternatively, provide an
                 // array to specify the transitions to make available in
                 // the transitions panel, for example...
                 // ['None', 'Fade', 'Slide']
                 transitions: false,

                 // Adds a menu button to the slides to open the menu panel.
                 // Set to 'false' to hide the button.
                 openButton: true,

                 // If 'true' allows the slide number in the presentation to
                 // open the menu panel. The reveal.js slideNumber option must
                 // be displayed for this to take effect
                 openSlideNumber: false,

                 // If true allows the user to open and navigate the menu using
                 // the keyboard. Standard keyboard interaction with reveal
                 // will be disabled while the menu is open.
                 keyboard: true,

                 // Normally the menu will close on user actions such as
                 // selecting a menu item, or clicking the presentation area.
                 // If 'true', the sticky option will leave the menu open
                 // until it is explicitly closed, that is, using the close
                 // button or pressing the ESC or m key (when the keyboard
                 // interaction option is enabled).
                 sticky: false,

                 // If 'true' standard menu items will be automatically opened
                 // when navigating using the keyboard. Note: this only takes
                 // effect when both the 'keyboard' and 'sticky' options are enabled.
                 autoOpen: true,

                 // If 'true' the menu will not be created until it is explicitly
                 // requested by calling RevealMenu.init(). Note this will delay
                 // the creation of all menu panels, including custom panels, and
                 // the menu button.
                 delayInit: false,

                 // If 'true' the menu will be shown when the menu is initialised.
                 openOnInit: false,

                 // By default the menu will load it's own font-awesome library
                 // icons. If your presentation needs to load a different
                 // font-awesome library the 'loadIcons' option can be set to false
                 // and the menu will not attempt to load the font-awesome library.
                 loadIcons: true
             },

             // CSS properties that can be auto-animated. Position & scale
             // is matched separately so there's no need to include styles
             // like top/right/bottom/left, width/height or margin.
             autoAnimateStyles: [
                 "opacity",
                 "color",
                 "background-color",
                 "padding",
                 "font-size",
                 "line-height",
                 "letter-spacing",
                 "border-width",
                 "border-color",
                 "border-radius",
                 "outline",
                 "outline-offset"
             ],

             help: true,

             autoAnimate: false,

             transition: "slide",

             backgroundTransition: 'zoom',

             katex: {
                 local: "node_modules/katex"
             }
         });
        </script>
    </body>
</html>
